opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/TimeZoneData.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.tasks.contract.TaskContract;
+
+import java.util.TimeZone;
+
+
+/**
+ * {@link RowData} for the time zone value of the task.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TimeZoneData implements RowData<TaskContract.Tasks>
```

Comment:
Setting time zone alone doesn't make sense. Since it's part of `DateTime` it should be set by the other date & time `RowData` implementations.

Reply:
Ideally OpenTasksPal provides high-level access to the content provider. Check out `SingleEventData` in CalendarPal it takes `DateTime`s and automatically sets timestamps, time zones and all-day flags. Loading or setting a time zone or an all-day flag without loading or setting the timestamp too doesn't make much sense and is considered error prone.

Reply:
There is only one timezone and all-day flag for tasks, I assume `start` and `due` with different timezone should be rejected, right? What about the all-day flag? Take it only from start, or validate if its the same for `start` and `due`?
Does it make sense to set all-day flag in `DueTime` from due?

Reply:
No different time zones should not be rejected. Actually we should update the provider to support this case. For now just shift start to the due time zone.
But start and due must have the same all-day flag.

Reply:
(addressed)